ui/app/components/identity/popup-members.js

```diff
@@ -3,37 +3,44 @@
  * SPDX-License-Identifier: BUSL-1.1
  */
 
-import { alias } from '@ember/object/computed';
-import { computed } from '@ember/object';
-import Base from './_popup-base';
-
-export default Base.extend({
-  model: alias('params.firstObject'),
-
-  groupArray: computed('params', function () {
-    return this.params.objectAt(1);
-  }),
-
-  memberId: computed('params', function () {
-    return this.params.objectAt(2);
-  }),
-
-  messageArgs(/*model, groupArray, memberId*/) {
-    return [...arguments];
-  },
-
-  successMessage(model, groupArray, memberId) {
-    return `Successfully removed '${memberId}' from the group`;
-  },
-
-  errorMessage(e, model, groupArray, memberId) {
-    const error = e.errors ? e.errors.join(' ') : e.message;
-    return `There was a problem removing '${memberId}' from the group - ${error}`;
-  },
-
-  transaction(model, groupArray, memberId) {
-    const members = model.get(groupArray);
-    model.set(groupArray, members.without(memberId));
-    return model.save();
-  },
-});
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import errorMessage from 'vault/utils/error-message';
+
+export default class IdentityPopupMembers extends Component {
+  @service flashMessages;
+  @tracked showConfirmModal = false;
+
+  onSuccess(memberId) {
+    if (this.args.onSuccess) {
+      this.args.onSuccess();
+    }
+    this.flashMessages.success(`Successfully removed '${memberId}' from the group`);
+  }
+  onError(err, memberId) {
+    if (this.args.onError) {
+      this.args.onError();
+    }
+    const error = errorMessage(err);
+    this.flashMessages.danger(`There was a problem removing '${memberId}' from the group - ${error}`);
+  }
+
+  transaction() {
+    const members = this.args.model[this.args.groupArray];
+    this.args.model[this.args.groupArray] = members.without(this.args.memberId);
```

Comment:
This is more for @chelshaw as Claire and I have already chatted about this. But I figured this might be a decent example of where I'm confused about what we _should not_ be allowed to do in a glimmer component and what we _can do_. Technically `this.args` are `readonly` and we should hit an error when we try and reassign them? —[article](https://v5.chriskrycho.com/journal/understanding-args-in-glimmer-components/) from Chris Krycho who talks about it in detail. Any idea how we're able to do this, I know we do it elsewhere in the app?

Reply:
It would throw an error if an arg itself was being reassigned, for example `this.args.model = null`. In this case a property on the arg is mutated. Maybe not best practice but acceptable from a framework standpoint.

Reply:
Jordan is right -- here we're not updating an arg, we're updating an attribute on an arg. Small but important difference!